### PR TITLE
Jhkim/issue 464

### PR DESCRIPTION
**Pull request by jhkimnew**
https://github.com/microsoft/IIS.WebManager/issues/464

1.	IIS.WebManager uses JSON.stringify() to serialize the JSON object of request information. For the File System API Put scenario, we used ArrayBuffer to send request body and that caused the issue to the JSON.stringify. 
It simply does not support the data type and body data was lost because of the issue. So, as a work-around, I have updated the data type from ArrayBuffer to Uint8Array. 
2.	For the File System API Get scenario, we serialized the respond body with byte[] array and that caused the data loss while deserializing if the file content is not pure UTF8 data. As a work-around, we should read the response body in string, not in byte[], to support various encoding such as UTF16 and UTF8, even UTF8 BOM characters. 
3.	In uploading scenario, there is a bug while handling the file attribute of lastUpdatedDate. We incorrectly used the number value instead of the supported ISO Date format string value. I have converted the number value with ISO Date format string value.

However, there is one remaining issue regarding downloading. The download seems not to be possible in the current design. We send the downlevel request to the backend server and the downlevel is started from the backend sever, not from the user computer. 
Because there is no easy fix for the downloading issue, I suggest that we should disable the downloading button for this time and consider to fix in the next milestone.

### CODE REVIEW

File: src/app/resources/scripts/iis_scripts/Invoke-LocalHttp.ps1 [hunk 1]
```diff
@@ -67,8 +67,19 @@ $uri = $uriBuilder.ToString()
...
 try {
     $httpMsg = New-Object System.Net.Http.HttpRequestMessage -ArgumentList $httpMethod, $uri
-    if ($reqObj._body) {
-        $requestContent = New-Object System.Net.Http.StringContent ([string] $reqObj._body)
+    if ($reqObj._body -or $reqObj._body_Uint8Array) {
+        if ($reqObj._body_Uint8Array) {
+          #
+          # convert $reqObj._body_Uint8Array to ByteArray(byte[])
+          #
+          $byteArray = new-object byte[] $reqObj._body_Uint8Array_Length
```
Comment (shhsu) [inline src/app/resources/scripts/iis_scripts/Invoke-LocalHttp.ps1]: `New-Object` for consistency

File: src/app/resources/scripts/iis_scripts/Invoke-LocalHttp.ps1 [hunk 2]
```diff
@@ -67,8 +67,19 @@ $uri = $uriBuilder.ToString()
...
+          for ($i = 0; $i -lt $reqObj._body_Uint8Array_Length; $i++) {
+            $byteArray[$i] = $reqObj._body_Uint8Array.$i;
```
Comment (shhsu) [inline src/app/resources/scripts/iis_scripts/Invoke-LocalHttp.ps1]: $reqObj._body_Uint8Array[$i] for consistency

**Review by shhsu — COMMENTED** [review]
Did we need to also encode the body on the way back?
Have you tried the download functionalities to download a binary file

File: src/app/resources/scripts/iis_scripts/Invoke-LocalHttp.ps1 (hunk 1 above)
Comment (jhkimnew) [reply src/app/resources/scripts/iis_scripts/Invoke-LocalHttp.ps1]: Good catch. Fixed.

File: src/app/resources/scripts/iis_scripts/Invoke-LocalHttp.ps1 [hunk 3]
```diff
@@ -135,4 +151,4 @@ if ($result) {
     $result
 } else {
     throw "Unexpected error occured"
-}
+}
```
Comment (shhsu) [inline src/app/resources/scripts/iis_scripts/Invoke-LocalHttp.ps1]: nit: newline

File: src/app/runtime/wac/services/powershell-service.ts [hunk 4]
```diff
@@ -132,4 +142,4 @@ export class PowershellService {
       }),
     );
   }
-}
+}
```
Comment (shhsu) [inline src/app/runtime/wac/services/powershell-service.ts]: nit: newline

**Review by shhsu — APPROVED** [review]
add back newlines and remove download button

File: src/app/resources/scripts/iis_scripts/Invoke-LocalHttp.ps1 (hunk 3 above)
Comment (jhkimnew) [reply src/app/resources/scripts/iis_scripts/Invoke-LocalHttp.ps1]: fixed

**Pull request merged**